### ISSUE TITLE
SurfaceTool.generate_ normals cannot be used normally

### ISSUE THREAD

**Issue by 1419427247** [godotengine/godot#65076]
### Godot version

v4.0.alpha14.official [106b68050]

### System information

Windows10,Vulkan

### I tried to generate a mesh by using surfacetool, but there was a problem with the normal of the generated mesh
**This causes me to manually call the function 'set_normal' to generate normals correctly**
![image](https://user-images.githubusercontent.com/33598732/187402970-ef5ca5d3-5942-43ea-a559-c171bc732965.png)

### In the same way that I applied in Godot 3.5, I generated the meshcorrectly
![屏幕截图 2022-08-30 173326](https://user-images.githubusercontent.com/33598732/187403850-2268a3f5-a9c4-4633-b88b-c583845b9c05.png)

### Steps to reproduce

1. Add a meshinstance3d node
2. Write the following code:

```gdscript
extends MeshInstance3D

const _brick_vertexs = [
	Vector3(0.5, -0.5, -0.5),
	Vector3(0.5, 0.5, -0.5),
	Vector3(-0.5, 0.5, -0.5),
	Vector3(-0.5, -0.5, -0.5),
	Vector3(0.5, -0.5, 0.5),
	Vector3(0.5, 0.5, 0.5),
	Vector3(-0.5, 0.5, 0.5),
	Vector3(-0.5, -0.5, 0.5),
]

const _brick_uvs = [
	Vector2(0,0),Vector2(1,0),Vector2(1,1),
	Vector2(0,0),Vector2(1,1),Vector2(0,1),
]

func _ready():
	var array_mesh = ArrayMesh.new()
	var surface_tool = SurfaceTool.new()
	surface_tool.begin(Mesh.PRIMITIVE_TRIANGLES)

	for x in range(32):
		for y in range(32):
			var up_vertexs = [
				_brick_vertexs[2],_brick_vertexs[1],_brick_vertexs[5],
				_brick_vertexs[2],_brick_vertexs[5],_brick_vertexs[6],
			]
			for i in range(6):
				surface_tool.set_uv(_brick_uvs[i])
				surface_tool.add_vertex(up_vertexs[i] + Vector3(x,0,y))
			
			var right_vertexs = [
				_brick_vertexs[2],_brick_vertexs[6],_brick_vertexs[7],
				_brick_vertexs[2],_brick_vertexs[7],_brick_vertexs[3],
			]
			for i in range(6):
				surface_tool.set_uv(_brick_uvs[i])
				surface_tool.add_vertex(right_vertexs[i] + Vector3(x,0,y))

	surface_tool.generate_normals()
	surface_tool.generate_tangents()
	surface_tool.commit(array_mesh)
	mesh = array_mesh
```

### Minimal reproduction project

[SurfaceToolBug.zip](https://github.com/godotengine/godot/files/9451821/SurfaceToolBug.zip)

**Comment by Calinou** [comment]
PS: Code blocks should use triple backticks like this (with an optional language name for syntax highlighting):

\`\`\`gdscript
code here
\`\`\`

I edited your post accordingly, but remember to do this in the future :slightly_smiling_face:

**Comment by Mathis-Z** [comment]
in 4.0 SurfaceTool currently generates all normals "smooth". In 3.5 you can have smooth normals generated by calling add_smooth_group() on the SurfaceTool similar to add_uv() or add_normal():
[https://docs.godotengine.org/de/latest/classes/class_surfacetool.html#class-surfacetool-method-add-smooth-group](https://docs.godotengine.org/de/latest/classes/class_surfacetool.html#class-surfacetool-method-add-smooth-group)

(Actually I tested that and it seems like this function is broken in 3.5stable as well)

In 4.0 however this function is even more broken as it just sets the smooth_group property of Vertex which is not even used in SurfaceTool but all normals are generated smooth anyways.
As a quick fix one could just generate all normals "un-smooth".

**Comment by akien-mga** [comment]
Fixed by #68034.
